app/views/applications/show.html.erb

```diff
@@ -11,6 +11,13 @@
 <% @application.pets.each do |pet| %>
   <section id="pet-<%= pet.id %>">
     <%= link_to(pet.name, "/pets/#{pet.id}") %>
-    - <%= link_to("Approve Application", "/applications/#{@application.id}/pets/#{pet.id}", method: :patch) %>
+    <% if pet.status == "Adoptable" %>
+      - <%= link_to("Approve Application", "/applications/#{@application.id}/pets/#{pet.id}?approval=approve", method: :patch) %>
+    <% elsif pet.status == "Pending" %>
+      - Pending Adoption
+      <% if @application.application_pets.find_by(pet_id: pet.id).approved %>
```

Comment:
if the application_pet corresponding to this show page's application and current pet in the enumerable has a status of approved, then the `Cancel Approval` link will appear. otherwise it won't